### DISCUSSION TITLE
Ignoring body of non-empty request results in TCP RST on esp32

### DISCUSSION THREAD

**Discussion by wesley800** [miguelgrinberg/microdot#159] [General]
I'm trying to implement a simple web switch on esp32c3, and used POST as the request method. I found that if comment out the `request.body` line, the client will get a immediate RST after HTTP status line. No idea if it's bug in network stack of esp-idf. Should we mention it somewhere in document?

```
@app.post("/switchpin")
def switch_IO(request):
    # do something
    request.body # simply consume the request body, if any
    return ("DONE", 200, {"Content-Type": "text/html; charset=UTF-8"})
```

**Comment by miguelgrinberg** [comment]
What is the client sending in the body of this request? I think another solution would be for the client to not send a body.

**Reply by wesley800** [reply]
If the submit button of a form on webpage has name attribute set, it should send its text on button as value, though the value is rarely used in reality.

**Reply by miguelgrinberg** [reply]
Yes, but if you are expecting a body it is normally required that the body is consumed, even if you don't need it. Microdot does not check for this, the application is responsible of consuming the body.
